olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -151,6 +151,47 @@ client::CancellationToken VersionedLayerClientImpl::GetData(
                             request.GetPriority());
 }
 
+client::CancellationToken VersionedLayerClientImpl::GetPartition(
+    TileRequest tile_request, PartitionsResponseCallback callback) {
+  auto data_task =
+      [=](client::CancellationContext context) mutable -> PartitionsResponse {
+    if (!tile_request.GetTileKey().IsValid()) {
+      return {{client::ErrorCode::InvalidArgument, "Tile key is invalid"}};
+    }
+
+    // FIXME: OnlineOnly since it probably possible needed quad tree could be
+    //  found in the cache without additional fields
+    const auto fetch_option = FetchOptions::OnlineOnly;
```

Comment:
But then we probably do not need it as we already downloaded the bundle also.
The fallback in this case should be different, means if we have something in cache, but it does not contain all the  requested fields we download it again with the requested fields and replace it in cache.